haystack/nodes/query_classifier/sklearn.py

```diff
@@ -5,15 +5,22 @@
 import urllib
 
 from tqdm.auto import tqdm
-from sklearn.ensemble._gb_losses import BinomialDeviance
-from sklearn.ensemble._gb import GradientBoostingClassifier
 
 from haystack.nodes.query_classifier.base import BaseQueryClassifier
 
 
 logger = logging.getLogger(__name__)
 
 
+try:
+    from sklearn.ensemble._gb_losses import BinomialDeviance
+    from sklearn.ensemble._gb import GradientBoostingClassifier
+except ImportError as exc:
+    logger.debug("sklearn could not be imported. Run 'pip install farm-haystack[metrics]' to fix this issue.")
```

Comment:
Similar to TfidfVectorizer, it doesn't make much sense from the user perspective that they need to install `metrics` for QueryClassifier. And scikit-learn is not in the `metrics` anyway. It's part of the standard dependencies.